docs/user_manual/performance-guide.md

```diff
@@ -6,6 +6,57 @@ SPDX-License-Identifier: MPL-2.0
 
 # Guidelines for performance enhancement
 
+The `power-grid-model` is a library that shines in its ability to handle calculations at scale.
+It remains performant, even when doing calculations with one or a combination of the following extremes (non-exhaustive):
+
+- Large grids
+- Batch calculations with many scenarios
+- Many changes in the grid in each scenario
+
+To achieve that high performance, several optimizations to the code are made.
+To use those optimizations to their fullest extend, we recommend our users to follow the following guidelines.
+
+## Data validity
+
+Many of our optimizations rely on assuming input data validity and the fact that the provided grid is reasonably close to realistic.
+Non-convergence, underdetermined equations (sparse matrices) or other unexpected behavior may therefore be encountered when the data is not realistic.
+
+To keep the PGM performant, checks on hard physical bounds are offloaded to a separate tool, the [data validator](data-validator.md).
+However, these checks are extremely expensive and should therefore not be used in production environments at scale when performance matters.
+Instead, we recommend using the data validator specifically for debugging purposes.
+
+```{note}
+Some combinations of input data are not forbidden by physics, but still pose unrealistic conditions, e.g. a source with a very low short-circuit power.
+These cases may result in unexpected behavior of the calculation core under the optimizations in the code.
+Vagueness and case-dependence make it hard to check what can be considered "unrealistic", and the [data validator](data-validator.md) will therefore not catch such cases.
+We recommend our users to provide reasonably realistic scenarios to prevent these edge cases from happening.
+```
+
+## Data format
+
+The data format of input, output and update data can have a big effect on memory and computational cost.
+
+### Input/update data volume
+
+Row-based data (created, e.g., using {py:class}`power_grid_model.initialize_array` in Python) constructs input/update data with all attributes for a given dataset type.
+However, many component attributes are optional.
+If your use case does not depend on these attributes, this means that a lot of data is needlessly created and initialized.
+If you are running on a system on which memory is the bottle-neck, using a columnar data format may reduce the memory burden, at the cost of a slight computational overhead during the calculations.
+
+### Output data volume
+
+For most use cases, only certain output values are relevant.
+For instance, if you are only interested in line loading, outputting all other components and attributes results in unnecessary overhead.
+The output data may be a significant, if not the dominant, contributor to memory load, particularly when running batch calculations with many scenarios.
```

Comment:
```suggestion
The output data may be a significant, if not the dominant, contributor to memory load, particularly when running batch calculations.
```

Reply:
I think _with many scenarios_ should be kept. It is very different to do 10 or 10000 scenarios.

Reply:
Then those numbers should appear. "many scenarios" and "batch calculations" are redundant. Either do "large batch calculations" or "batch calculations with millions of scenarios"

Reply:
_Large batch calculations_ sounds indeed better and removes redundancy.

Reply:
right.
```suggestion
The output data may be a significant, if not the dominant, contributor to memory load, particularly when running large batch calculations.
```